src/datatypes/structures.js

```diff
@@ -84,7 +84,16 @@ function writeContainer(value, buffer, offset, typeArgs, rootNode) {
       debug(new Error("Missing Property " + typeArgs.fields[index].name).stack);
       console.log(context);
     }
-    offset = this.write(context[typeArgs.fields[index].name], buffer, offset, typeArgs.fields[index], rootNode);
+    try {
+      offset = this.write(context[typeArgs.fields[index].name], buffer, offset, typeArgs.fields[index], rootNode);
+    }
+    catch(err)
+    {
+      console.log("typeArgs : " + JSON.stringify(typeArgs.fields[index]));
+      console.log("context : "+JSON.stringify(context));
+      console.log("value : " + JSON.stringify(context[typeArgs.fields[index].name]));
+      throw err;
+    }
```

Comment:
Having something like https://github.com/roblabla/Protocols/blob/master/lib/protocols.js#L98-L105 would help to get more debug information here. The structural fields (Structure and Array) could use it to add context as to where the error occured in the fieldInfo chain.